### ISSUE TITLE
proposal: Rename ConnManager

### ISSUE THREAD

**Issue by Wondertan** [libp2p/go-libp2p#3033]
Reading through #3032, I thought of the looming confusion between the meanings of "manager" and "tracker." Also, the word "manager" is too broad for what the component actually does. Thus, the proposal is to rename it. The best and only option I have is `ConnTrimmer`

**Comment by sukunrt** [comment]
I'm fine with `ConnTrimmer`. Another alternative is `ConnGC`.

**Comment by Stebalien** [comment]
I agree but also, please be careful and avoid any massive breaking changes (e.g., leave an alias behind and make sure the change doesn't break much).
